python/pyarrow/io.pxi

```diff
@@ -1257,6 +1283,10 @@ cdef class BufferReader(NativeFile):
     cdef:
         Buffer buffer
 
+    # XXX Needed to make numpydoc happy
+    def __init__(self, obj):
```

Comment:
Do you remember why this was needed? (which error did it give?)

Aside, we should probably try to instruct sphinx/autodoc/numpydoc to not include `__init__` functions in the reference docs, as I don't think we have any case where this adds value (compared to the class docstring). See eg https://arrow.apache.org/docs/dev/python/generated/pyarrow.parquet.ParquetDataset.html#pyarrow.parquet.ParquetDataset.__init__

Reply:
The problem is that numpydoc tries to match the parameters documented in the `BufferReader` docstring with its `__init__` signature. To be honest, I'm not sure why some classes prefer to define `__cinit__` rather than `__init__`, there doesn't seem to be a consistent convention across the codebase.

Reply:
Opened https://issues.apache.org/jira/browse/ARROW-16123 for this